Remove ADD_SOURCE_WEBCORE_DERIVED_DEPENDENCIES macro

mcatanzaro opened the pull request:
#### 72ddfd39b7bdf4997b026e96e9208d1517bb2535
<pre>
Remove ADD_SOURCE_WEBCORE_DERIVED_DEPENDENCIES macro
<a href="https://bugs.webkit.org/show_bug.cgi?id=266984">https://bugs.webkit.org/show_bug.cgi?id=266984</a>

Reviewed by Fujii Hironori and Don Olmstead.

ADD_SOURCE_WEBCORE_DERIVED_DEPENDENCIES is used in two cases:

(1) To add dependencies on generated header files. In most cases, these
headers are already part of WebCore_PRIVATE_FRAMEWORK_HEADERS, and so
are guaranteed to be generated before compiling WebCore. In all other
cases, they should just be added to WebCore_PRIVATE_FRAMEWORK_HEADERS.

(2) To add dependencies on generated source files. In all cases, these
sources are already part of WebCore_SOURCES, and so are guaranteed to be
generated before compiling WebCore.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCoreMacros.cmake:

Canonical link: <a href="https://commits.webkit.org/272974@main">https://commits.webkit.org/272974@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f6e21cdb86787c09ee53b2d610a6550693c43ba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33674 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12446 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35604 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36290 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/30542 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/34721 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14820 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9601 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29625 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34150 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10495 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30019 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9161 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9255 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30029 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37614 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30563 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30352 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35386 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9396 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7345 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33271 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11174 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7800 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9979 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10172 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont reviewed (commented):
Where's the actual removal of the macro?

donny-dont reviewed (approved):
@fujii is there any Visual Studio issue if we remove this? Looks fine otherwise.

mcatanzaro commented on Source/WebCore/Sources.txt:
```diff
@@ -3144,6 +3144,10 @@ xml/parser/XMLDocumentParserLibxml2.cpp
+// FIXME: These files are individually added to WebCore_SOURCES (as they must be in order to ensure
+// they are generated before WebCore is compiled). They should not be listed here as this means they
+// are compiled twice, violating the one-definition rule. But I don't know how to remove them
+// without breaking XCode builds.
```
This comment is wrong. The files are only being built once (unified). I'm trying to figure out why.

mcatanzaro replied:
tbh I'm going to give up now. I don't understand.

I'll just remove the FIXME because the right thing is happening. Everything's fine.

mcatanzaro replied:
OK, I found my mistake. The source files added to WebCore_SOURCES do not get compiled because `WEBKIT_COMPUTE_SOURCES()` is smart and marks them all as header files if they're included in the unified build, so their presence in WebCore_SOURCES only creates a dependency to guarantee they're generated in time and does not cause them to be compiled separately.

Sorry for spreading so much confusion.

mcatanzaro committed:
Remove ADD_SOURCE_WEBCORE_DERIVED_DEPENDENCIES macro
https://bugs.webkit.org/show_bug.cgi?id=266984

Reviewed by Fujii Hironori and Don Olmstead.

ADD_SOURCE_WEBCORE_DERIVED_DEPENDENCIES is used in two cases:

(1) To add dependencies on generated header files. In most cases, these
headers are already part of WebCore_PRIVATE_FRAMEWORK_HEADERS, and so
are guaranteed to be generated before compiling WebCore. In all other
cases, they should just be added to WebCore_PRIVATE_FRAMEWORK_HEADERS.

(2) To add dependencies on generated source files. In all cases, these
sources are already part of WebCore_SOURCES, and so are guaranteed to be
generated before compiling WebCore.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCoreMacros.cmake:

Canonical link: https://commits.webkit.org/272974@main
